monkey/infection_monkey/network/tools.py

```
@@ -159,7 +159,7 @@ def check_tcp_ports(ip, ports, timeout=DEFAULT_TIMEOUT, get_banner=False):
             if get_banner and (len(connected_ports_sockets) != 0):
                 readable_sockets, _, _ = select.select([s[1] for s in connected_ports_sockets], [], [], 0)
                 # read first BANNER_READ bytes
-                banners = [sock.recv(BANNER_READ).decode() if sock in readable_sockets else ""
+                banners = [sock.recv(BANNER_READ).decode(errors='ignore') if sock in readable_sockets else ""
```

Please document why we ignore errors here